File: java/src/main/java/deltix/luminary/LiteralDuration.java
package deltix.luminary;

public class LiteralDuration extends Literal {
    private String value;

    public LiteralDuration(String value) {
        super(LiteralKind.DURATION);
        this.value = value;
    }

    public String getValue() {
        return value;
    }

    @Override
    public String toString() {
        return "\"" + value + "\"";
    }

    public static final String MIN_VALUE_AS_STRING = "#MIN";
    public static final String MAX_VALUE_AS_STRING = "#MAX";
}

File: csharp/main/LiteralDuration.cs
using System;

namespace Deltix.Luminary
{
	public sealed class LiteralDuration : Literal, ILiteral<String>
	{
		public const String MinValueAsString = "#MIN";
		public const String MaxValueAsString = "#MAX";

		public LiteralDuration(String value)
			: base(LiteralKind.Duration)
		{
			Value = value;
		}

		public String Value { get; }

		public override String ToString()
		{
			return $"\"{Value}\"";
		}
	}
}
